src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#[macro_use]
+extern crate lazy_static;
+extern crate rand;
+
+use rand::Rng;
+use rand::distributions::{Normal, IndependentSample};
+
+///
+/// A rand::Rng implementation which is safe for wasm-unknown-unknown.
+///
+/// Requires `javascript_math_random` to be an import identifier corresponding
+/// to the browser's JavaScript Math.random function.
+///
+pub struct JavaScriptRng;
+
+extern "C" {
+    fn javascript_math_random() -> f64;
+}
+
+impl Rng for JavaScriptRng {
+    fn next_u32(&mut self) -> u32 {
+        unsafe {
+            // The maximum value of a u32 is 4,294,967,295. We go one over here
+            // because the JavaScript Math.random call returns values between
+            // 0 (inclusive) and 1 (exclusive). Without the + 1 here, the floor
+            // call would mean the maximum u32 value couldn't be returned.
+            let upper: i64 = 4294967295 + 1;
+            let deviate: f64 = javascript_math_random() * (upper as f64);
+
+            deviate.floor() as u32
```

Comment:
I really feel ill-equipped to review anything that has to do with cryptography, even if it's `Math.random` which isn't even meant to be cryptographically secure. One stupid off-by-one and you're toast.